### ISSUE TITLE
does not exist and causes issue with numpy 2.0 release

### ISSUE THREAD

**Issue by naruto-raj** [wandb/wandb#7812]
https://github.com/wandb/wandb/blob/4d70e448675cab1c0ee0fb882ebff6f45490a489/wandb/sdk/data_types/_dtypes.py#L393

would it be better to remove it?

**Comment by LeanderK** [comment]
this is a major issue for me and caused all my CI to crash. I would like my project to be forward compatible with the latest numpy but this is not possible with the current wandb dependency!

**Comment by QuyAnh2005** [comment]
@LeanderK Try to reinstall numpy with version 1.26.4 `pip install numpy==1.26.4`

**Comment by LeanderK** [comment]
@QuyAnh2005 I have done that everywhere but I consider that a short-term fix and unfortunate that the only reason for this is my dependency on `wandb`.  `wandb` should fix their code (and should have already done so as being deep in the tool chain!)

**Comment by naruto-raj** [comment]
We can either lock 1.26.4 as dependency or remove np.float_ with a PR.

**Comment by LeanderK** [comment]
Thank you, I realised that I sounded a bit harsh, this has caused me a bit of an headache. I think it would be the correct way to resolve this quickly, as wandb is a logging framework and it should not come in the way of choosing the right numpy version. Lot's of smaller packages already have bound constraints and a logging framework should always try to be compatible, as it's mixed and matched with a lot of other code.
